src/components/ActionList/ActionList.tsx

```diff
@@ -30,7 +30,7 @@ export default function ActionList({
   }
 
   const hasMultipleSections = finalSections.length > 1;
-  const Element: string = hasMultipleSections ? 'ul' : 'div';
+  const Element: any = hasMultipleSections ? 'ul' : 'div';
```

Comment:
I'm surprised we can't type this as `keyof HTMLElementTagNameMap` or `Element` or something similar. Do you think this is because of something in our tsconfig or a version of a types package?

Reply:
I tried those and had no luck.

When doing `<ul></ul>` it says the type is of `JSX.Element` so I tried doing a `keyof JSX.Element` (which resolves to an array of strings containing ul and div) but I still got an error.

It looks like the root cause of this is detailed at https://github.com/Microsoft/TypeScript/issues/28768. It seems this was fixed in typescript 3.3 (so you don't need any casting at all), but we can't move onto that till our linting tooling is ready :(

Reply:
We could change the return line to the following, but I think that fix is worse than changing the typing:

```
return React.createElement(
    Element,
    {className: styles.ActionList},
    ...sectionMarkup,
  );
```

Reply:
If it's fixed in a later version of typescript I'm fine with using `any` 😄  We should just leave a note something so we don't forget about it.